src/main/java/org/radarcns/webapp/ManagementPortalEndPoint.java

```diff
@@ -0,0 +1,222 @@
+package org.radarcns.webapp;
+/*
+ * Copyright 2016 King's College London and The Hyve
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.radarcns.managementportal.MpClient;
+import org.radarcns.managementportal.Project;
+import org.radarcns.managementportal.Subject;
+import org.radarcns.webapp.util.ResponseHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import static org.radarcns.webapp.util.BasePath.PROJECT;
+import static org.radarcns.webapp.util.BasePath.SUBJECTS;
+import static org.radarcns.webapp.util.Parameter.STUDY_ID;
+import static org.radarcns.webapp.util.Parameter.SUBJECT_ID;
+
+/**
+ *  Management Portal web-app. Function set to access subject and source information from MP.
+ *  A subject is a person enrolled for in a study. A source is a device linked to the subject.
+ */
+@Api
+@Path("/" + "mp")
+public class ManagementPortalEndPoint {
```

Comment:
perhaps it already does, but it might be good to mimic the new API in management portal here, as merged from RADAR-CNS/ManagementPortal#101.

Reply:
Yes I think it already does, the requests for subject are based on login and the requests for project are based on project names.